coderd/prometheusmetrics/prometheusmetrics.go

```diff
@@ -72,36 +74,43 @@ func ActiveUsers(ctx context.Context, registerer prometheus.Registerer, db datab
 }
 
 // Workspaces tracks the total number of workspaces with labels on status.
-func Workspaces(ctx context.Context, registerer prometheus.Registerer, db database.Store, duration time.Duration) (func(), error) {
+func Workspaces(ctx context.Context, logger slog.Logger, registerer prometheus.Registerer, db database.Store, duration time.Duration) (func(), error) {
 	if duration == 0 {
-		duration = 5 * time.Minute
+		duration = defaultRefreshRate
 	}
 
-	gauge := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	workspaceStatuses := prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: "coderd",
 		Subsystem: "api",
 		Name:      "workspace_latest_build_total",
-		Help:      "The latest workspace builds with a status.",
+		Help:      "The current number of workspace builds by status.",
 	}, []string{"status"})
-	err := registerer.Register(gauge)
-	if err != nil {
+	if err := registerer.Register(workspaceStatuses); err != nil {
+		return nil, err
+	}
+
+	workspaceDetails := prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "coderd",
+		Subsystem: "api",
+		Name:      "workspace_detail",
```

Comment:
Naming: can we just call it `coder_workspaces`? We have `coderd_agents_apps`, `coderd_insights_parameters`. I wouldn't pair with `api` specifically.

EDIT:

Following the Prometheus naming convention (credits to @spikecurtis for linking in a review a while ago), we could also consider `coder_workspace_info`.

Reply:
Regarding the `api` subsystem, I just copied how the other metric defined in this function is named. I agree that bit is probably extraneous and am happy to take it out.

`_info` metrics are usually reserved for "static" metrics, such as `build_info` and things like that.

Reply:
> Regarding the api subsystem, I just copied how the other metric defined in this function is named. I agree that bit is probably extraneous and am happy to take it out.

👍 

> _info metrics are usually reserved for "static" metrics, such as build_info and things like that.

TIL! Thanks for the explanation.